src/snek5000/output/history_points.py

```diff
@@ -0,0 +1,115 @@
+from pathlib import Path
+from io import StringIO
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+class HistoryPoints:
+    """History points (Nek5000 output)
+
+    See https://nek5000.github.io/NekDoc/problem_setup/features.html#history-points
+
+    """
+
+    def __init__(self, output=None):
+        self.output = output
+        sim = output.sim
+        params = sim.params
+        hp_params = params.output.history_points
+
+        self.coords = hp_params.coords
+
+        if self.coords is None:
+            return
+
+        self.nb_points = len(self.coords)
+        self.coords = np.array(self.coords)
+        shape = self.coords.shape
+
+        if shape[0] >= params.oper.max.hist or shape[1] != params.oper.dim:
+            raise ValueError(
+                "Shape of params.output.history_points.coords are not compatible."
+                f"Should be (< {params.oper.max.hist = }, == {params.oper.dim = })"
+            )
+
+        # temporary?
+        output.path_session.mkdir(exist_ok=True)
+
+        self.path_file = output.path_session / f"{sim.info_solver.short_name}.his"
+        if self.path_file.exists():
+            return
+
+        with open(self.path_file, "w") as file:
```

Comment:
Add a check for `self.output._has_to_save` here before writing.

Reply:
Done!